academic_observatory_workflows/workflows/orcid_telescope.py

```diff
@@ -0,0 +1,756 @@
+
+    def cleanup(self, release: OrcidRelease, **kwargs) -> None:
+        """Delete all files, folders and XComs associated with this release."""
+        cleanup(dag_id=self.dag_id, execution_date=kwargs["logical_date"], workflow_folder=release.workflow_folder)
+
+
+def create_orcid_batch_manifest(
+    orcid_batch: OrcidBatch, reference_date: pendulum.DateTime, bucket: str, bucket_prefix: str = None
+) -> None:
+    """Create a manifest (.csv) for each orcid batch containing blob names and modification dates of all the
+    modified files in the bucket's orcid directory. Only blobs modified after the reference date are included.
+
+    :param orcid_directory: The OrcidBatch instance for this orcid directory
+    :param reference_date: The date to use as a reference for the manifest
+    :param bucket: The name of the bucket
+    :param bucket_prefix: The prefix to use when listing blobs in the bucket. i.e. where the orcid directories are located
+    :return: The path to the date manifest, the path to the uri manifest
+    """
+    prefix = f"{bucket_prefix}/{orcid_batch.batch_str}/" if bucket_prefix else f"{orcid_batch.batch_str}/"
+
+    logging.info(f"Creating manifests for {orcid_batch.batch_str}")
+    blobs = gcs_list_blobs(bucket, prefix=prefix)
+    manifest = []
+    for blob in blobs:
+        if pendulum.instance(blob.updated) >= reference_date:
+            manifest.append(
+                {
+                    MANIFEST_HEADER[0]: blob.bucket.name,
+                    MANIFEST_HEADER[1]: blob.name,
+                    MANIFEST_HEADER[2]: blob.updated,
+                }
+            )
+
+    with open(orcid_batch.manifest_file, "w", newline="") as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=MANIFEST_HEADER)
+        writer.writeheader()
+        writer.writerows(manifest)
+
+    logging.info(f"Manifest saved to {orcid_batch.manifest_file}")
+
+
+def latest_modified_record_date(manifest_file_path) -> pendulum.DateTime:
+    """Reads the manifest file and finds the most recent date of modification for the records
+
+    :param manifest_file_path: the path to the manifest file
+    :return: the most recent date of modification for the records
+    """
+    with open(manifest_file_path, "r") as f:
+        reader = csv.DictReader(f)
+        modified_dates = sorted([pendulum.parse(row["updated"]) for row in reader])
+    return modified_dates[-1]
+
+
+def orcid_batch_names() -> List[str]:
+    """Create a list of all the possible ORCID directories
+
+    :return: A list of all the possible ORCID directories
+    """
+    n_1_2 = [str(i) for i in range(10)]  # 0, 1, 2, 3, 4, 5, 6, 7, 8, 9
+    n_3 = n_1_2.copy() + ["X"]  # 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, X
+    combinations = list(itertools.product(n_1_2, n_1_2, n_3))  # Creates the 000 to 99X directory structure
+    return ["".join(i) for i in combinations]
+
+
+def transform_orcid_record(record_path: str) -> Union(Dict, str):
```

Comment:
Should use square brackets with Union:
```suggestion
def transform_orcid_record(record_path: str) -> Union[Dict, str]:
```

Reply:
Sure, is this just a convention thing?

Reply:
Yeah it is just how type hints are written, e.g. https://docs.python.org/3/library/typing.html#typing.Union